File: Models/Authentication/Password/ForgotPassword.cs
using System.ComponentModel.DataAnnotations;

namespace aliment_backend.Models.Authentication.Password
{
    /// <summary>
    /// Représente les données nécessaires pour demander la réinitialisation de mot de passe.
    /// </summary>
    public class ForgotPassword
    {
        /// <summary>
        /// Obtient ou définit l'adresse e-mail associée au compte utilisateur pour la réinitialisation du mot de passe.
        /// </summary>
        [EmailAddress(ErrorMessage = "Veuillez saisir une adresse e-mail valide.")]
        [Required(ErrorMessage = "L'adresse e-mail est requise.")]
        public string? Email { get; set; }
    }

}

File: src/app/features/auth/_models/forgotPassword.ts
/**
 * L'interface `ForgotPassword` est utilisée pour représenter les données nécessaires à la perte d'un mot de passe.
 * Elle est généralement utilisée dans le cadre d'une fonctionnalité de récupération de mot de passe.
 */
export interface ForgotPassword {
  /**
   * Adresse e-mail du destinataire pour la récupération du mot de passe.
   * @type {string}
   */
  email: string
}
